greynoise_connector.py

```diff
@@ -182,92 +245,86 @@ def _check_apikey(self, action_result):
+                    license_type=license_type,
+                    expiration=expiration))
                 self.save_progress("Your licence is expired and therefore your API key has community permissions")
-                self.debug_print("Validated API Key. License type: {license_type}, Expiration: {expiration}".format(license_type=license_type, expiration=expiration))
+                self.debug_print("Validated API Key. License type: {license_type}, Expiration: {expiration}".format(
+                    license_type=license_type,
+                    expiration=expiration))
                 self.debug_print("Your licence is expired and therefore your API key has community permissions")
                 return phantom.APP_SUCCESS
             else:
-                self.save_progress("Validated API Key. License type: {license_type}, Expiration: {expiration}".format(license_type=license_type, expiration=expiration))
-                self.debug_print("Validated API Key. License type: {license_type}, Expiration: {expiration}".format(license_type=license_type, expiration=expiration))
+                self.save_progress("Validated API Key. License type: {license_type}, Expiration: {expiration}".format(
+                    license_type=license_type,
+                    expiration=expiration))
+                self.debug_print("Validated API Key. License type: {license_type}, Expiration: {expiration}".format(
+                    license_type=license_type,
+                    expiration=expiration))
                 return phantom.APP_SUCCESS
         else:
             self.save_progress("Invalid response from API")
             try:
                 response_json = json.dumps(response_json)
-            except:
+            except Exception:
                 return action_result.set_status(
                     phantom.APP_ERROR, "Invalid response from API"
                 )
             return action_result.set_status(
                 phantom.APP_ERROR, "Invalid response from API: %s" % response_json
             )
 
-    def _test_connectivity(self, param):
-        action_result = self.add_action_result(ActionResult(dict(param)))
-        ret_val = self._check_apikey(action_result)
-        if phantom.is_fail(ret_val):
-            self.save_progress("Test Connectivity Failed")
-            return ret_val
+    # check the config for the type of license to determine which actions can be used
+    def _check_license_type(self):
+        config = self.get_config()
 
-        self.save_progress("Test Connectivity Passed")
-        return action_result.set_status(phantom.APP_SUCCESS)
+        license_type = "enterprise"
+        message = "Enterprise license."
 
-    def _lookup_ip(self, param):
-        action_result = self.add_action_result(ActionResult(dict(param)))
-        ret_val = self._check_apikey(action_result)
-        if phantom.is_fail(ret_val):
-            return ret_val
+        if(config.get("license_type") == "community"):
```

Comment:
```suggestion
        if config.get("license_type") == "community":
```

same for any other cases with parentheses around `if`/`while` statements

Reply:
Done